numpy/core/src/multiarray/datetime.c

```diff
@@ -3822,18 +3822,26 @@ recursive_find_object_timedelta64_type(PyObject *obj,
                  * single object using [()], but not by using
                  * __getitem__(integer) approaches
                  */
-                PyObject *item, *meth, *args;
+                PyObject *item, *args;
 
-                meth = PyObject_GetAttrString(obj, "__getitem__");
-                args = Py_BuildValue("(())");
-                item = PyObject_CallObject(meth, args);
+                args = PyTuple_New(0);
+                if (args == NULL) {
+                    return 0;
+                }
+                item = PyObject_GetItem(obj, args);
+                Py_DECREF(args);
+                if (item == NULL) {
+                    return 0;
+                }
```

Comment:
The code above does not seem super pretty, but I am not quite sure that subclasses are not supposed to get a chance here, so `PyObject_GetItem` is no behaviour change I think. Oddly the `datetime` code looks a bit different than the `timedelta` one above.